src/Management/src/Endpoint/Actuators/LogFile/LogFileEndpointHandler.cs

```diff
@@ -0,0 +1,57 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Steeltoe.Management.Configuration;
+
+namespace Steeltoe.Management.Endpoint.Actuators.Logfile;
+
+public sealed class LogFileEndpointHandler : ILogFileEndpointHandler
+{
+    private readonly IOptionsMonitor<LogFileEndpointOptions> _optionsMonitor;
+    private readonly ILogger<LogFileEndpointHandler> _logger;
+
+    public EndpointOptions Options => _optionsMonitor.CurrentValue;
+
+    public LogFileEndpointHandler(IOptionsMonitor<LogFileEndpointOptions> optionsMonitorMonitor, ILoggerFactory loggerFactory)
+    {
+        ArgumentNullException.ThrowIfNull(optionsMonitorMonitor);
+        ArgumentNullException.ThrowIfNull(loggerFactory);
+
+        _optionsMonitor = optionsMonitorMonitor;
+        _logger = loggerFactory.CreateLogger<LogFileEndpointHandler>();
+    }
+
+    public async Task<string> InvokeAsync(object? argument, CancellationToken cancellationToken)
```

Comment:
If this returned `Task<string?>`, the middleware could respond with 404 when `null` is returned by overriding `WriteResponseAsync`. Should catch a `FileNotFoundException` here to detect this. This is normal behavior; some logging frameworks don't create the file until there's something to log, so it shouldn't log a warning. But when the configured path is `null`, the endpoint was turned on without configuring a path, which is a mistake, so a warning should be logged in that case (and `null` should be returned).